static/app/components/carousel.tsx

```diff
@@ -0,0 +1,140 @@
+import {useEffect, useRef, useState} from 'react';
+import styled from '@emotion/styled';
+
+import {Button} from 'sentry/components/button';
+import {IconArrow} from 'sentry/icons';
+
+type Props = {
+  children?: React.ReactNode;
+};
+
+function Carousel({children}: Props) {
+  const ref = useRef<HTMLDivElement | null>(null);
+  const [isAtStart, setIsAtStart] = useState<boolean>(true);
+  const [isAtEnd, setIsAtEnd] = useState<boolean>(false);
+  const [showArrows, setShowArrows] = useState<boolean>(true);
+
+  const setStartAndEnd = (scrollLeft: number, totalScrollDist: number) => {
+    // scrollLeft: the amount scrolled horizontally.
+    // start at 0 and end at totalScrollDist
+
+    // totalScrollDist = scrollWidth - clientWidth
+    // scrollWidth: total element width including overflow
+    // clientWidth: width of the element excluding overflow
+    if (scrollLeft < 10) {
+      setIsAtStart(true);
+      setIsAtEnd(false);
+    } else if (scrollLeft > totalScrollDist - 10) {
+      setIsAtStart(false);
+      setIsAtEnd(true);
+    } else {
+      setIsAtStart(false);
+      setIsAtEnd(false);
+    }
+  };
+
+  useEffect(() => {
+    if (!ref.current) {
+      return;
+    }
+    const contentWidth = ref.current.scrollWidth - ref.current.clientWidth;
+    if (contentWidth === 0) {
+      setShowArrows(false);
+    }
+
+    const scrollLeft = ref.current.scrollLeft;
+    setStartAndEnd(scrollLeft, contentWidth);
+  }, [ref]);
+
+  const scroll = (direction: string) => {
+    requestAnimationFrame(() => {
+      if (!ref.current) {
+        return;
+      }
+
+      const scrollLeft = ref.current.scrollLeft;
+      const itemWidth = parseInt(getComputedStyle(ref.current.children[0]).width, 10);
+
+      let scrollDist = 0;
+      if (direction === 'left') {
+        scrollDist = scrollLeft - itemWidth;
+      } else if (direction === 'right') {
+        scrollDist = scrollLeft + itemWidth;
+      }
+      ref.current.scrollLeft = scrollDist;
+    });
+  };
+
+  const setArrows = () => {
+    requestAnimationFrame(() => {
+      if (!ref.current) {
+        return;
+      }
+
+      const scrollLeft = ref.current.scrollLeft;
+      const totalScrollDist = ref.current.scrollWidth - ref.current.clientWidth;
+
+      setStartAndEnd(scrollLeft, totalScrollDist);
+    });
+  };
+
+  return (
+    <CarouselContainer>
+      <CarouselItems ref={ref} onScroll={setArrows}>
+        {children}
+      </CarouselItems>
+      {showArrows && !isAtStart && (
+        <CircledArrow onClick={() => scroll('left')} direction="left" />
+      )}
+      {showArrows && !isAtEnd && (
+        <CircledArrow onClick={() => scroll('right')} direction="right" />
+      )}
+    </CarouselContainer>
+  );
+}
+
+const CarouselContainer = styled('div')`
+  position: relative;
+`;
+
+const CarouselItems = styled('div')`
+  display: flex;
+  overflow-x: scroll;
+  scroll-behavior: smooth;
```

Comment:
Probably need some extra CSS to style the scrollbar, otherwise it will look bad on Windows.